sympy/sets/sets.py

```diff
@@ -377,6 +377,28 @@ def is_subset(self, other):
             # This assumes that an unevaluated Intersection will always come
             # back as an Intersection...
             elif isinstance(s_o, Intersection):
+                from sympy import ConditionSet
+                # TODO: add logic for Interval, Union...
+                if isinstance(self, ConditionSet) and isinstance(other, ConditionSet):
+                    # Check whether one condition is a subset of the other condition
+                    interval_s = self.condition.as_set().intersect(self.base_set)
+                    interval_o = other.condition.as_set().intersect(other.base_set)
```

Comment:
What does this do for something like this:
```julia
In [24]: s = ConditionSet((x, y), x<1, Reals*Reals)                                                                                                           

In [25]: s                                                                                                                                                    
Out[25]: 
⎧                   2        ⎫
⎨(x, y) | (x, y) ∊ ℝ  ∧ x < 1⎬
⎩                            ⎭

In [26]: s.condition.as_set().intersect(s.base_set)                                                                                                           
Out[26]: 
          ⎛ 2⎞
(-∞, 1) ∩ ⎝ℝ ⎠
```
That last intersection should ideally be the empty set.

Reply:
Also:
```julia
In [29]: ConditionSet(x, x<y, Reals)                                                                                                                          
Out[29]: {x | x ∊ ℝ ∧ x < y}

In [30]: ConditionSet(x, x<y, Reals).condition.as_set()                                                                                                       
---------------------------------------------------------------------------
NotImplementedError 
```

Reply:
Regarding `as_set` not being implemented for multivariate expressions, I realize that, what would you suggest to do in this PR about that?

Reply:
> That last intersection should ideally be the empty set.

Yes, but isn't that also out of scope for this PR? I could just `pass` or `return None` if `s.sym` has more than a single symbol

Reply:
I'm asking what this PR does for is_subset in those cases and how that compares with current behaviour on master. You don't need to fix everything in one PR but we should try not to introduce new wrong results.

Given
```julia
In [1]: s1 = ConditionSet((x, y), x<1, Reals*Reals)                                                                                                           

In [2]: s2 = ConditionSet((x, y), x>1, Reals*Reals)                                                                                                           

In [3]: s1.is_subset(s2)
```
the two intersections above *should* result in the empty set. If they did evaluate to the empty set then the logic you add here would return True which would be incorrect.